store/src/java-test/com/zimbra/cs/imap/ImapServerListenerPoolTest.java

```diff
@@ -0,0 +1,118 @@
+package com.zimbra.cs.imap;
+
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertSame;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.zimbra.client.ZMailbox;
+import com.zimbra.common.account.Key;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.cs.account.Account;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.mailbox.MailboxTestUtil;
+public class ImapServerListenerPoolTest {
+    private Provisioning prov = Provisioning.getInstance();
+    @BeforeClass
+    public static void init() throws Exception {
+        MailboxTestUtil.initProvisioning();
+    }
+
+    @Test
+    public void testInstanceManagement() throws ServiceException {
+        String serverName1 = "example1.zimbra.com";
```

Comment:
@fatkudu as far as I can tell, there is nothing to clean up, because everything is created in memory in maps in MockProvisioning instance, so all this data should be gone when test exits.